rclcpp/CMakeLists.txt

```diff
@@ -10,8 +10,8 @@ find_package(rmw_implementation REQUIRED)
 find_package(rmw_implementation_cmake REQUIRED)
 find_package(rosidl_generator_cpp REQUIRED)
 
-if(NOT WIN32)
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wall -Wextra")
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  set_directory_properties(PROPERTIES COMPILE_OPTIONS "-std=c++11;-Wall;-Wextra;-Wpedantic")
```

Comment:
Changing the condition is meant to only add these flags to compilers that support these flags. Clang is included explicitly, because that is a targeted compiler that understands Wpedantic.
I used set_directory_properties, because I thought it was a little more precise. I switched over to using add_compile_options instead, because I thought it was clearer.